pkg/devfile/parser/parse.go

```diff
@@ -397,11 +400,24 @@ func parseFromURI(importReference v1.ImportReference, curDevfileCtx devfileCtx.D
 func parseFromRegistry(importReference v1.ImportReference, resolveCtx *resolutionContextTree, tool resolverTools) (d DevfileObj, err error) {
 	id := importReference.Id
 	registryURL := importReference.RegistryUrl
+	destDir := "."
+
+	// contains the downloaded stack from the registry
+	tempDir, _ := ioutil.TempDir("", "")
+	defer os.RemoveAll(tempDir)
+
 	if registryURL != "" {
-		devfileContent, err := getDevfileFromRegistry(id, registryURL, importReference.Version)
+		err = getStackFromRegistry(id, registryURL, tempDir, destDir)
```

Comment:
to locate & download a specific version from registry, the `Version` information is also required. https://github.com/devfile/api/blob/351f05b7c2b18e8a449dad444029cdf5e1ca81cc/pkg/apis/workspaces/v1alpha2/import_reference.go#L60